### PR TITLE
CVSB-10751 Added multiple tech-records selection page

### PR DESCRIPTION
**Pull request by agoncearenco**
This change depends on 10754 and 10742.
I had to split the `searchVehicle` functionality and moved it from the vehicle-lookup page to the vehicle-service

### CODE REVIEW

File: src/providers/vehicle/vehicle.service.ts [hunk 1]
```diff
@@ -52,12 +65,43 @@ export class VehicleService {
     this.visitService.updateVisit();
   }
 
-  getVehicleTechRecord(param, searchCriteria): Observable<HttpResponse<VehicleTechRecordModel>> {
-    return this.httpService.getTechRecords(param, searchCriteria);
+  getVehicleTechRecords(searchedValue: string, searchCriteriaQueryParam: string): Observable<VehicleModel[]> {
+    searchedValue = searchedValue.replace(/\s+/g, '');
+    return this.httpService.getTechRecords(searchedValue.toUpperCase(), searchCriteriaQueryParam)
+      .pipe(
```
Comment (KHodderDVSA) [inline src/providers/vehicle/vehicle.service.ts]: strange choice to use pipe - map returns a new object which can be directly "map"ed again.

Comment (agoncearenco) [reply src/providers/vehicle/vehicle.service.ts]: RxJS says that they are functionally equivalent and suggest to use the [pipeable operators](https://stackoverflow.com/a/47552482), that's why I also added it in the `getTestResultsHistory` method. If it introduces confusion, I can remove

File: src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          content_type: 'error',
+          item_id: "Failed retrieving the testResultsHistory"
+        });
+        this.storageService.update(STORAGE.TEST_HISTORY, []);
+        this.goToVehicleDetails(selectedVehicle);
+      },
+      complete: function () {}
+    };
+
+    this.vehicleService.getTestResultsHistory(selectedVehicle.systemNumber).subscribe(testHistoryResponseObserver).add(()=>{
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: what does .add do? does something more after the observable has completed?

File: src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts [hunk 3]
```diff
@@ -23,19 +23,20 @@ import * as logsActions from "../../../../modules/logs/logs.actions";
...
+import {MultipleTechRecordsSelectionPage} from './multiple-tech-records-selection/multiple-tech-records-selection';
 
 @IonicPage()
 @Component({
   selector: 'page-vehicle-lookup',
   templateUrl: 'vehicle-lookup.html'
 })
 export class VehicleLookupPage {
-  testData: TestModel;
+  combinationTestData: TestModel;
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts]: why is this called combination test? it won't always be a combinationTest

Comment (agoncearenco) [reply src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts]: You're right. Didn't notice that. Will rename

File: src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts (hunk 2 above)
Comment (agoncearenco) [reply src/pages/testing/vehicle/vehicle-lookup/multiple-tech-records-selection/multiple-tech-records-selection.ts]: It adds a [teardown logic](http://reactivex.io/rxjs/class/es6/Subscription.js~Subscription.html#instance-method-add) to this subscription.
Basically I did this to dismiss the loading when the observable has completed its execution ( either successfully or with an error).

File: src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts (hunk 3 above)
Comment (agoncearenco) [reply src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts]: Done

File: src/providers/vehicle/vehicle.service.ts (hunk 1 above)
Comment (KHodderDVSA) [reply src/providers/vehicle/vehicle.service.ts]: I think it does add confusion, but honestly at this point...

Comment (agoncearenco) [reply src/providers/vehicle/vehicle.service.ts]: I'll remove

Comment (agoncearenco) [reply src/providers/vehicle/vehicle.service.ts]: Done

**Pull request closed without merging**